src/main/java/seedu/address/logic/commands/FetchBillCommand.java

```diff
@@ -13,17 +20,68 @@
 
     public static final String COMMAND_WORD = "bill";
 
-    public static final String MESSAGE_SUCCESS = "This is the bill for the guest.";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Retrieves the bill of a guest.\n"
+            + "Parameters:"
+            + PREFIX_ID + "PERSONID "
+            + "[" + PREFIX_ROOMNUMBER + "ROOMNUMBER]...\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_ID + "J0000000 "
+            + PREFIX_ROOMNUMBER + "101";
+
+    public static final String MESSAGE_SUCCESS_NONSPECIFIC = "These are the bill/s for guest %1$s (%2$s).";
+    public static final String MESSAGE_SUCCESS_SPECIFIC = "This is the bill for room %1$s of guest %2$s (%3$s).";
+    public static final String MESSAGE_GUEST_NONEXISTENT = "Guest %1$s does not exist in the system.";
+    public static final String MESSAGE_ROOM_NONEXISTENT = "Room %1$s does not exist in the system.";
+
+    private final PersonId personId;
+    private final boolean isSpecific;
+    private final String roomNum;
 
     /**
-     *
+     * @param personId to fetch the bills of
      */
-    public FetchBillCommand() {
-        //requireNonNull(person);
+    public FetchBillCommand(PersonId personId) {
+        requireNonNull(personId);
+        this.personId = personId;
+        this.roomNum = null;
+        this.isSpecific = false;
+    }
+
+    /**
+     * Specifically fetch the bill of room number {@code roomNum} for {@code personId}
+     */
+    public FetchBillCommand(PersonId personId, String roomNum) {
+        requireNonNull(personId);
+        requireNonNull(roomNum);
+        this.personId = personId;
+        this.roomNum = roomNum;
+        this.isSpecific = true;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        return new CommandResult(String.format(MESSAGE_SUCCESS));
+        requireNonNull(model);
+
+        Optional<Person> person = model.findPersonWithId(personId);
+
+        if (person.isEmpty()) {
+            throw new CommandException(MESSAGE_GUEST_NONEXISTENT);
+        }
+
+        if (isSpecific) {
+            Optional<Room> room = model.findRoom(personId.toString());
```

Comment:
model.findRoom takes roomNum as parameters. We haven't made findRoomByPersonId() for now. Could you either create the method in the model manager or change this part?